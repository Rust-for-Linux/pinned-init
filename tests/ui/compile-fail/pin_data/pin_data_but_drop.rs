use pin_init::*;

#[pin_data]
struct Foo {
    a: usize,
}

impl Drop for Foo {
    fn drop(&mut self) {}
}

fn main() {}
