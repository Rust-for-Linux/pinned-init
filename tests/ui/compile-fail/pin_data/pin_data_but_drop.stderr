error[E0119]: conflicting implementations of trait `MustNotImplDrop` for type `Foo`
 --> tests/ui/compile-fail/pin_data/pin_data_but_drop.rs:3:1
  |
3 | #[pin_data]
  | ^^^^^^^^^^^
  | |
  | first implementation here
  | conflicting implementation for `Foo`
  |
  = note: this error originates in the attribute macro `pin_data` (in Nightly builds, run with -Z macro-backtrace for more info)
